### ISSUE TITLE
Error when starting: ERROR: [Errno 48] Address already in use

### ISSUE THREAD

**Issue by pulpopurpura** [localstack/localstack#1872]
I've just started using localstack. I downloaded the latest version, started localstack with 'localstack start' in one terminal, and entered 'localstack web' in another.  I got the following error

```
2019-12-17T17:18:12:WARNING:bootstrap.py: Unable to load plugins from file /usr/local/lib/python3.7/site-packages/localstack_ext/plugins.py: No module named 'localstack_ext.utils.aws.aws_models'
ERROR: [Errno 48] Address already in use
```

I'm running Python 3.7, on MacOS 10.14.  I restarted the machine but same result.  One thing is that when I first went to run 'localstack web' I got a series of errors that Python modules like 'modo' , 'flask_cors' and 'amazon_kclpy' weren't installed so (not knowing Python) I tried to install them using 'pip3 install amazon_kclpy', etc.

I tried uninstalling localhost 'npm uninstall global localhost', and reinstalling it, thinking perhaps that would install whatever was missing, but that didn't help.

**Comment by whummer** [comment]
Thanks for reporting @pulpopurpura . Can you please post the log output of this command?
```
DEBUG=1 localstack start
```
Do you have Docker installed? Also, are there any Docker containers running that take up certain network ports? (`docker ps`). Thanks

**Comment by tomevari** [comment]
I encountered the same issues on a fresh pip install of localstack (0.11.1.1) running `localstack web`, had to install flask, moto and a handful of other libraries using pip to get it to start. Once I did, it throws `[Errno 48] Address already in use`.

I'm using Python 3.7 and MacOS 10.15.1.

The only docker container I have running is localstack itself.
```
docker ps 
CONTAINER ID        IMAGE                   COMMAND                  CREATED             STATUS              PORTS                                                                                                    NAMES
dc676cca20fc        localstack/localstack   "docker-entrypoint.sh"   29 minutes ago      Up 29 minutes       0.0.0.0:4567-4620->4567-4620/tcp, 0.0.0.0:8080-8081->8080-8081/tcp, 4566/tcp, 0.0.0.0:10000->10000/tcp   localstack_main
```

The full output of `DEBUG=1 localstack web`
```
DEBUG=1 localstack web
2020-05-21T16:55:01:DEBUG:bootstrap.py: Loading plugins - scope "commands", module "localstack_ext": <function register_localstack_commands at 0x110c84320>
LocalStack version: 0.11.1.1
2020-05-21T16:55:06:DEBUG:bootstrap.py: Loading plugins - scope "services", module "localstack_ext": <function register_localstack_plugins at 0x1107d4f80>
2020-05-21T16:55:06:DEBUG:bootstrap.py: Loading plugins - scope "services", module "localstack": <function register_localstack_plugins at 0x110c844d0>
Unable to register plugins: cannot import name 'MethodNotFoundException' from 'moto.apigateway.exceptions' (/usr/local/lib/python3.7/site-packages/moto/apigateway/exceptions.py)
2020-05-21T16:55:06:WARNING:bootstrap.py: Unable to load plugins from file /usr/local/lib/python3.7/site-packages/localstack/plugins.py: cannot import name 'MethodNotFoundException' from 'moto.apigateway.exceptions' (/usr/local/lib/python3.7/site-packages/moto/apigateway/exceptions.py)
2020-05-21T16:55:06:ERROR:localstack.services.generic_proxy: Exception running proxy on port 8081: [Errno 48] Address already in use Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/localstack/services/generic_proxy.py", line 427, in run_cmd
    self.httpd = ThreadedHTTPServer((self.listen_host, self.port), GenericProxyHandler)
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/socketserver.py", line 452, in __init__
    self.server_bind()
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/server.py", line 137, in server_bind
    socketserver.TCPServer.server_bind(self)
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/socketserver.py", line 466, in server_bind
    self.socket.bind(self.server_address)
OSError: [Errno 48] Address already in use

Traceback (most recent call last):
  File "/usr/local/bin/localstack", line 90, in main
    subcommand['function'](argv, args)
  File "/usr/local/lib/python3.7/site-packages/localstack/utils/cli.py", line 57, in cmd_web
    localstack.dashboard.api.serve(port)
  File "/usr/local/lib/python3.7/site-packages/localstack/dashboard/api.py", line 149, in serve
    generic_proxy.serve_flask_app(app=app, port=port, quiet=True)
  File "/usr/local/lib/python3.7/site-packages/localstack/services/generic_proxy.py", line 476, in serve_flask_app
    app.run(port=int(port), threaded=True, host=host, ssl_context=ssl_context)
  File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 990, in run
    run_simple(host, port, self, **options)
  File "/usr/local/lib/python3.7/site-packages/werkzeug/serving.py", line 1052, in run_simple
    inner()
  File "/usr/local/lib/python3.7/site-packages/werkzeug/serving.py", line 1005, in inner
    fd=fd,
  File "/usr/local/lib/python3.7/site-packages/werkzeug/serving.py", line 848, in make_server
    host, port, app, request_handler, passthrough_errors, ssl_context, fd=fd
  File "/usr/local/lib/python3.7/site-packages/werkzeug/serving.py", line 740, in __init__
    HTTPServer.__init__(self, server_address, handler)
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/socketserver.py", line 452, in __init__
    self.server_bind()
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/http/server.py", line 137, in server_bind
    socketserver.TCPServer.server_bind(self)
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/socketserver.py", line 466, in server_bind
    self.socket.bind(self.server_address)
OSError: [Errno 48] Address already in use
ERROR: [Errno 48] Address already in use
```

And `DEBUG=1 localstack start` (though localstack is otherwise working fine):
```
2020-05-21T16:57:42:DEBUG:bootstrap.py: Loading plugins - scope "commands", module "localstack_ext": <function register_localstack_commands at 0x105eb1320>
LocalStack version: 0.11.1.1
Starting local dev environment. CTRL-C to quit.
2020-05-21T16:57:42:DEBUG:bootstrap.py: Loading plugins - scope "services", module "localstack_ext": <function register_localstack_plugins at 0x105a01f80>
2020-05-21T16:57:42:DEBUG:bootstrap.py: Loading plugins - scope "services", module "localstack": <function register_localstack_plugins at 0x105eb14d0>
docker run -it -e DEFAULT_REGION="us-east-1" -e LOCALSTACK_HOSTNAME="localhost" -e DEBUG="1" -e TEST_AWS_ACCOUNT_ID="000000000000" -p 4566:4566 --rm --privileged --name localstack_main -p 8080-8081:8080-8081 -p 4567-4620:4567-4620 -p 10000:10000  -v "/private/var/folders/gs/rxqns63d107d9lsmdv_4_kt00000gp/T/localstack:/tmp/localstack" -v "/var/run/docker.sock:/var/run/docker.sock" -e DOCKER_HOST="unix:///var/run/docker.sock" -e HOST_TMP_FOLDER="/private/var/folders/gs/rxqns63d107d9lsmdv_4_kt00000gp/T/localstack" "localstack/localstack" 
Waiting for all LocalStack services to be ready
2020-05-21 08:57:44,467 CRIT Supervisor is running as root.  Privileges were not dropped because no user is specified in the config file.  If you intend to run as root, you can set user=root in the config file to avoid this message.
2020-05-21 08:57:44,475 INFO supervisord started with pid 14
2020-05-21 08:57:45,478 INFO spawned: 'dashboard' with pid 20
2020-05-21 08:57:45,482 INFO spawned: 'infra' with pid 21
(. .venv/bin/activate; bin/localstack web)
2020-05-21 08:57:45,491 INFO success: dashboard entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
(. .venv/bin/activate; exec bin/localstack start --host)
2020-05-21T08:57:46:DEBUG:bootstrap.py: Loading plugins - scope "commands", module "localstack_ext": <function register_localstack_commands at 0x7f61d89cb4c0>
LocalStack version: 0.11.1
2020-05-21 08:57:46,635 INFO success: infra entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2020-05-21T08:57:46:DEBUG:bootstrap.py: Loading plugins - scope "services", module "localstack_ext": <function register_localstack_plugins at 0x7f61d8dfa550>
2020-05-21T08:57:46:DEBUG:bootstrap.py: Loading plugins - scope "services", module "localstack": <function register_localstack_plugins at 0x7f61d89cb550>
LocalStack version: 0.11.1
Starting local dev environment. CTRL-C to quit.
2020-05-21T08:57:46:DEBUG:bootstrap.py: Loading plugins - scope "commands", module "localstack_ext": <function register_localstack_commands at 0x7f6f68e2d4c0>
2020-05-21T08:57:46:DEBUG:bootstrap.py: Loading plugins - scope "services", module "localstack_ext": <function register_localstack_plugins at 0x7f6f6925c550>
2020-05-21T08:57:46:DEBUG:bootstrap.py: Loading plugins - scope "services", module "localstack": <function register_localstack_plugins at 0x7f6f68e2d550>
Waiting for all LocalStack services to be ready
Starting edge router (https port 4566)...
Starting mock API Gateway service in http ports 4566 (recommended) and 4567 (deprecated)...
Starting mock CloudFormation service in http ports 4566 (recommended) and 4581 (deprecated)...
Starting mock CloudWatch service in http ports 4566 (recommended) and 4582 (deprecated)...
Starting mock DynamoDB service in http ports 4566 (recommended) and 4569 (deprecated)...
Starting mock DynamoDB Streams service in http ports 4566 (recommended) and 4570 (deprecated)...
Starting mock EC2 service in http ports 4566 (recommended) and 4597 (deprecated)...
Starting mock ES service in http ports 4566 (recommended) and 4578 (deprecated)...
Starting mock Firehose service in http ports 4566 (recommended) and 4573 (deprecated)...
Starting mock IAM service in http ports 4566 (recommended) and 4593 (deprecated)...
Starting mock STS service in http ports 4566 (recommended) and 4592 (deprecated)...
Starting mock Kinesis service in http ports 4566 (recommended) and 4568 (deprecated)...
Starting mock KMS service in http ports 4566 (recommended) and 4599 (deprecated)...
Starting mock Lambda service in http ports 4566 (recommended) and 4574 (deprecated)...
Starting mock CloudWatch Logs service in http ports 4566 (recommended) and 4586 (deprecated)...
2020-05-21T08:57:52:INFO:localstack.multiserver: Starting multi API server process on port 51492
2020-05-21 08:57:52,733:API:  * Running on http://0.0.0.0:4565/ (Press CTRL+C to quit)
2020-05-21 08:57:52,733:API:  * Running on http://0.0.0.0:4565/ (Press CTRL+C to quit)
2020-05-21 08:57:52,733:API:  * Running on http://0.0.0.0:4565/ (Press CTRL+C to quit)
2020-05-21 08:57:52,785:API:  * Running on http://0.0.0.0:4582/ (Press CTRL+C to quit)
2020-05-21 08:57:52,785:API:  * Running on http://0.0.0.0:4582/ (Press CTRL+C to quit)
2020-05-21 08:57:52,785:API:  * Running on http://0.0.0.0:4582/ (Press CTRL+C to quit)
2020-05-21 08:57:52,809:API:  * Running on http://0.0.0.0:4558/ (Press CTRL+C to quit)
2020-05-21 08:57:52,809:API:  * Running on http://0.0.0.0:4558/ (Press CTRL+C to quit)
2020-05-21 08:57:52,809:API:  * Running on http://0.0.0.0:4558/ (Press CTRL+C to quit)
Starting mock Redshift service in http ports 4566 (recommended) and 4577 (deprecated)...
INFO[2020-05-21 08:57:53.076] No file found at path /init/seed.yaml; skipping seeding.
INFO[2020-05-21 08:57:53.076] Data will be stored in /tmp/local-kms
INFO[2020-05-21 08:57:53.076] Local KMS started on 0.0.0.0:4554
Starting mock Route53 service in http ports 4566 (recommended) and 4580 (deprecated)...
Starting mock S3 service in http ports 4566 (recommended) and 4572 (deprecated)...
Starting mock Secrets Manager service in http ports 4566 (recommended) and 4584 (deprecated)...
Starting mock SES service in http ports 4566 (recommended) and 4579 (deprecated)...
Starting mock SNS service in http ports 4566 (recommended) and 4575 (deprecated)...
Starting mock SQS service in http ports 4566 (recommended) and 4576 (deprecated)...
Starting mock SSM service in http ports 4566 (recommended) and 4583 (deprecated)...
Starting mock Cloudwatch Events service in http ports 4566 (recommended) and 4587 (deprecated)...
Initializing DynamoDB Local with the following configuration:
Port:   4563
InMemory:       true
DbPath: null
SharedDb:       true
shouldDelayTransientStatuses:   false
CorsParams:     *
Listening at http://:::4564

Starting mock StepFunctions service in http ports 4566 (recommended) and 4585 (deprecated)...
Step Functions Local
Version: 1.5.0
Build: 2020-02-15
2020-05-21 08:57:55.305: Configure [Account] to [000000000000]
2020-05-21 08:57:55.315: Configure [Region] to [us-east-1]
2020-05-21 08:57:55.316: Configure [Lambda Endpoint] to [http://localhost:4574]
2020-05-21 08:57:55.317: Configure [DynamoDB Endpoint] to [http://localhost:4569]
2020-05-21 08:57:55.317: Configure [SQS Endpoint] to [http://localhost:4576]
2020-05-21 08:57:55.324: Configure [SNS Endpoint] to [http://localhost:4575]
2020-05-21 08:57:55.325: Configure [Step Functions Endpoint] to [http://localhost:4585]
2020-05-21 08:57:55.361: Loaded credentials from environment
2020-05-21 08:57:55.361: Starting server on port 8083 with account 000000000000, region us-east-1
08:57:56.898 [main] INFO  org.elasticmq.server.Main$ - Starting ElasticMQ server (0.15.5) ...
Waiting for all LocalStack services to be ready
May 21, 2020 8:57:58 AM com.amazonaws.http.apache.client.impl.ApacheConnectionManagerFactory createSocketFactoryRegistry
WARNING: SSL Certificate checking for endpoints has been explicitly disabled.
May 21, 2020 8:57:58 AM com.amazonaws.http.apache.client.impl.ApacheConnectionManagerFactory createSocketFactoryRegistry
WARNING: SSL Certificate checking for endpoints has been explicitly disabled.
May 21, 2020 8:57:58 AM com.amazonaws.http.apache.client.impl.ApacheConnectionManagerFactory createSocketFactoryRegistry
WARNING: SSL Certificate checking for endpoints has been explicitly disabled.
May 21, 2020 8:57:59 AM com.amazonaws.http.apache.client.impl.ApacheConnectionManagerFactory createSocketFactoryRegistry
WARNING: SSL Certificate checking for endpoints has been explicitly disabled.
08:57:59.299 [elasticmq-akka.actor.default-dispatcher-6] INFO  akka.event.slf4j.Slf4jLogger - Slf4jLogger started
May 21, 2020 8:57:59 AM com.amazonaws.http.apache.client.impl.ApacheConnectionManagerFactory createSocketFactoryRegistry
WARNING: SSL Certificate checking for endpoints has been explicitly disabled.
May 21, 2020 8:57:59 AM com.amazonaws.http.apache.client.impl.ApacheConnectionManagerFactory createSocketFactoryRegistry
WARNING: SSL Certificate checking for endpoints has been explicitly disabled.
May 21, 2020 8:57:59 AM com.amazonaws.http.apache.client.impl.ApacheConnectionManagerFactory createSocketFactoryRegistry
WARNING: SSL Certificate checking for endpoints has been explicitly disabled.
May 21, 2020 8:57:59 AM com.amazonaws.http.apache.client.impl.ApacheConnectionManagerFactory createSocketFactoryRegistry
WARNING: SSL Certificate checking for endpoints has been explicitly disabled.
May 21, 2020 8:57:59 AM com.amazonaws.http.apache.client.impl.ApacheConnectionManagerFactory createSocketFactoryRegistry
WARNING: SSL Certificate checking for endpoints has been explicitly disabled.
08:58:03.108 [main] INFO  org.elasticmq.server.Main$ - === ElasticMQ server (0.15.5) started in 8952 ms ===
08:58:03.110 [elasticmq-akka.actor.default-dispatcher-6] INFO  o.e.rest.sqs.TheSQSRestServerBuilder - Started SQS rest server, bind address 0.0.0.0:4560, visible server address http://localhost:4576
Ready.
```

Hope this is helpful!

**Comment by whummer** [comment]
Thanks for the update @tomevari  @pulpopurpura . This should be fixed in the `latest` version - a fix has been pushed in 4014831e4dfcd41b3ea81d267c48c121bee6d0f9 which should now allow you to customize `PORT_WEB_UI=...` or `START_WEB=0` in the main LocalStack container startup, which should resolve the port conflict you've reported. Hope that helps - please report here if the problem persists. Thanks
